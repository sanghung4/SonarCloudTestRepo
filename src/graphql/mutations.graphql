mutation purgeMincronCounts ($input: CountsInput!) {
  purgeMincronCounts (input: $input)
}

mutation addUser($userEmail: String!, $branchId: String!) {
  addUser(userEmail: $userEmail, branchId: $branchId) {
    success
    error {
      code
      message
        details
    }
  }
}

mutation removeUserBranch($userEmail: String!, $branchId: String!) {
  removeUserBranch(userEmail: $userEmail, branchId: $branchId) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation updateUserEmail($oldUserEmail: String!, $newUserEmail: String!) {
  updateUserEmail(oldUserEmail: $oldUserEmail, newUserEmail: $newUserEmail) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation deleteUser($userEmail: String!) {
  deleteUser(userEmail: $userEmail) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation SetSpecialPrices($input: PriceChangeInput!) {
  specialPrices(input: $input) {
    successfulUploads {
      uploadedPath
      uploadedName
    }
  }
}

mutation LoadCount($branchId: String!, $countId: String!) {
  loadCount(branchId: $branchId, countId: $countId) {
    id
    branchId
    countId
    branchName
    status
    errorMessage
    createdAt
    totalProducts
  }
}

mutation removeBranchCounts($input: RemoveBranchCountsInput!){
  removeBranchCounts(input: $input){
    deletedCounts
    {
      countLocations
      countLocationItems
      varianceCountLocationItemQuantities
      countLocationItemQuantities
      writeIns
    }
  }
}