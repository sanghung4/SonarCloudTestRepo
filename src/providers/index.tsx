import React from 'react';
import { ApolloProvider } from '@apollo/client';
import client from 'apollo/client';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import ThemeProvider from './Theme';
import AuthProvider from './Auth';
import OverlayProvider from './Overlay';
import { theme } from './Theme/constants';
import AppStateProvider from './AppState';

const Providers: React.FC = ({ children }) => (
  <ApolloProvider client={client}>
    <ThemeProvider theme={theme}>
      <OverlayProvider>
        <AuthProvider>
          <AppStateProvider>
          <SafeAreaProvider>{children}</SafeAreaProvider>
          </AppStateProvider>
        </AuthProvider>
      </OverlayProvider>
    </ThemeProvider>
  </ApolloProvider>
);

export default Providers;
