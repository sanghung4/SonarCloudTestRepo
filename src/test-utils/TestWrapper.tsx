import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
