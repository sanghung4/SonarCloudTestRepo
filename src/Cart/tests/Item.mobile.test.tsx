import { act } from '@testing-library/react';

import ItemMobile from 'Cart/Item.mobile';
import {
  mockCartContext,
  mockCartContract,
  mockCartProducts
} from 'Cart/tests/mocks';
import { LineItem } from 'generated/graphql';
import { CartContext } from 'providers/CartProvider';
import {
  actClickOnElement,
  actUpdateInput,
  updateInput
} from 'test-utils/commonFireEvents';
import { render } from 'test-utils/TestWrapper';

/**
 * Types
 */
type MockProps = {
  updateItemQuantity?: (
    itemId: string,
    quantity: number,
    minIncrementQty: number,
    productName: string
  ) => void;
  readOnly?: boolean;
  index?: number;
};

/**
 * Mock values
 */
const mocks = { cart: { ...mockCartContext } };

/**
 * Setup function
 */
function setup(m: typeof mocks, lineItem: LineItem, props?: MockProps) {
  return render(
    <CartContext.Provider value={m.cart}>
      <ItemMobile
        handleQuantityUpdate={jest.fn()}
        lineItem={lineItem}
        index={0}
        {...props}
      />
    </CartContext.Provider>
  );
}

/**
 * Test
 */
describe('Cart - Item.mobile', () => {
  afterEach(() => {
    mocks.cart = { ...mockCartContext };
  });

  it('Should match snapshot with product', () => {
    const lineItem = mockCartProducts[0];
    const { container } = setup(mocks, lineItem);
    act(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('Should match snapshot as readOnly', () => {
    const lineItem = mockCartProducts[0];
    const props = { readOnly: true };
    const { container } = setup(mocks, lineItem, props);
    act(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('Should match snapshot as contract with line notes', () => {
    const lineItem = mockCartProducts[0];
    mocks.cart.contract = mockCartContract;
    mocks.cart.lineNotes = { [lineItem.id!]: 'test' };
    const props = { readOnly: true };
    const { container } = setup(mocks, lineItem, props);
    act(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('Expect deleteItem is called when clicking on the delete button', () => {
    const lineItem = mockCartProducts[0];
    const { getByTestId } = setup(mocks, lineItem);
    actClickOnElement(getByTestId('remove-item-button-0'));
    expect(mocks.cart.deleteItem).toBeCalled();
  });

  it('Expect updateItemQuantity is called when changing the qty input', () => {
    const updateQtyFn = jest.fn();
    const lineItem = mockCartProducts[0];
    const props = { updateItemQuantity: updateQtyFn };
    const { getByTestId } = setup(mocks, lineItem, props);
    updateInput(getByTestId('quantity-input-0'), '20');
    act(() => {
      expect(updateQtyFn).toBeCalled();
    });
  });

  it('Expect updateItemQuantity is called when changing the qty input with no name', () => {
    const updateQtyFn = jest.fn();
    const lineItem = mockCartProducts[1];
    const props = { updateItemQuantity: updateQtyFn, index: 1 };
    const { getByTestId } = setup(mocks, lineItem, props);
    updateInput(getByTestId('quantity-input-1'), '20');
    act(() => {
      expect(updateQtyFn).toBeCalled();
    });
  });

  it('Expect modal save button to be in the document when opening line notes modal', async () => {
    const lineItem = mockCartProducts[0];
    mocks.cart.contract = mockCartContract;
    const { getByTestId } = setup(mocks, lineItem);
    actClickOnElement(getByTestId('cart-item-open-line-notes-0'));
    act(() => {
      const saveBtn = getByTestId('line-notes-modal-save');
      expect(saveBtn).toBeInTheDocument();
    });
  });

  it('Expect setLineNotes to be called when updating the line notes from modal', async () => {
    const testValue = 'test';
    const lineItem = mockCartProducts[0];
    mocks.cart.contract = mockCartContract;
    const { getByTestId } = setup(mocks, lineItem);

    actClickOnElement(getByTestId('cart-item-open-line-notes-0'));
    const input = getByTestId('line-notes-modal-textfield');
    actUpdateInput(input, testValue);
    actClickOnElement(getByTestId('line-notes-modal-save'));

    expect(input).toHaveDisplayValue(testValue);
    expect(mocks.cart.setLineNotes).toBeCalledWith({
      [lineItem.id!]: testValue
    });
  });

  it('Expect setLineNotes not to be called when the value is the same from updating the line notes from modal', async () => {
    const testValue = 'test';
    const lineItem = mockCartProducts[0];
    mocks.cart.contract = mockCartContract;
    mocks.cart.lineNotes = { [lineItem.id!]: testValue };
    const { getByTestId } = setup(mocks, lineItem);

    actClickOnElement(getByTestId('cart-item-open-line-notes-0'));
    const input = getByTestId('line-notes-modal-textfield');
    actUpdateInput(input, testValue);
    actClickOnElement(getByTestId('line-notes-modal-save'));

    expect(input).toHaveDisplayValue(testValue);
    expect(mocks.cart.setLineNotes).not.toBeCalled();
  });

  it('Expect qty and qtyAvailable to set as 0 when they are undefined', () => {
    const lineItem = {
      ...mockCartProducts[0],
      qtyAvailable: null,
      quantity: null
    };
    lineItem.product = {
      ...mockCartProducts[0].product!,
      minIncrementQty: null
    };
    const { getByTestId } = setup(mocks, lineItem);
    expect(getByTestId(`cart-item-qty-available-0`)).toHaveTextContent('0');
  });
});
