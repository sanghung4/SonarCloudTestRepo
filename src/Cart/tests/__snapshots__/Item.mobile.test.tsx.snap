// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart - Item.mobile Should match snapshot as contract with line notes 1`] = `
<div>
  <div
    class="MuiBox-root css-1iy8us4"
  >
    <div
      class="MuiBox-root css-nqdwa1"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-x3cpnl"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fysqgs-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                <img
                  alt="1/2\\" Wrot Copper Cap"
                  class="MuiBox-root css-idsjlm"
                  src="http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-kfrp3-MuiGrid-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-pad2pl-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1z0dvx8-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-rk7jif-MuiTypography-root"
            data-testid="cart-item-manufacturer-name-0"
          >
            Generic
          </span>
          <div>
            <p
              aria-label="1/2\\" Wrot Copper Cap"
              class="MuiTypography-root MuiTypography-body1 css-4mc1p7-MuiTypography-root"
              data-mui-internal-clone-element="true"
              data-testid="cart-item-product-name-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                1/2" Wrot Copper Cap
              </a>
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-zex3r6-MuiTypography-root"
              data-testid="cart-item-part-number-0"
            >
              Part#
               
              33860
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-qagaf3-MuiTypography-root"
              data-testid="cart-item-manufacturer-number-0"
            >
              MFR#
               
              CCAPD
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-18wiyal-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1d0yzg8"
          >
            <div
              class="MuiBox-root css-pori7h"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
              >
                Order Qty
              </span>
              <div
                class="MuiBox-root css-d0uhtl"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
                  data-testid="cart-item-order-qty-0"
                >
                  1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="test"
        class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-1t486iq-MuiGrid-root"
        data-mui-internal-clone-element="true"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-48hhhr-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-64pe8k-MuiTypography-root"
          >
            Line Notes
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true css-sqw2nl-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1hdqiad-MuiTypography-root"
          >
            test
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-2a4e9h-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-16lfl4z-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
            >
              Price
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
            data-testid="cart-item-unit-price-0"
          >
            $43.00 ea
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-1bay0si-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-6b5z9k-MuiTypography-root"
            >
              Order Total
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1fmbqzj-MuiTypography-root"
            data-testid="cart-item-total-price-0"
          >
            $43.00
          </h5>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiBox-root css-4rm8pb-MuiDivider-root"
    />
  </div>
</div>
`;

exports[`Cart - Item.mobile Should match snapshot as readOnly 1`] = `
<div>
  <div
    class="MuiBox-root css-1m9t0zg"
  >
    <div
      class="MuiBox-root css-nqdwa1"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-x3cpnl"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fysqgs-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                <img
                  alt="1/2\\" Wrot Copper Cap"
                  class="MuiBox-root css-idsjlm"
                  src="http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-kfrp3-MuiGrid-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-pad2pl-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1z0dvx8-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-rk7jif-MuiTypography-root"
            data-testid="cart-item-manufacturer-name-0"
          >
            Generic
          </span>
          <div>
            <p
              aria-label="1/2\\" Wrot Copper Cap"
              class="MuiTypography-root MuiTypography-body1 css-4mc1p7-MuiTypography-root"
              data-mui-internal-clone-element="true"
              data-testid="cart-item-product-name-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                1/2" Wrot Copper Cap
              </a>
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-zex3r6-MuiTypography-root"
              data-testid="cart-item-part-number-0"
            >
              Part#
               
              33860
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-qagaf3-MuiTypography-root"
              data-testid="cart-item-manufacturer-number-0"
            >
              MFR#
               
              CCAPD
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-18wiyal-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1d0yzg8"
          >
            <div
              class="MuiBox-root css-pori7h"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
              >
                Order Qty
              </span>
              <div
                class="MuiBox-root css-d0uhtl"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
                  data-testid="cart-item-order-qty-0"
                >
                  1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-2a4e9h-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-16lfl4z-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
            >
              Price
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
            data-testid="cart-item-unit-price-0"
          >
            $0.43 ea
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4 css-11ojwqs-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
            >
              Available Qty
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
            data-testid="cart-item-qty-available-0"
          >
            215
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ecxygc-MuiTypography-root"
          >
            in stock
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-1bay0si-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-6b5z9k-MuiTypography-root"
            >
              Order Total
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1fmbqzj-MuiTypography-root"
            data-testid="cart-item-total-price-0"
          >
            $0.43
          </h5>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiBox-root css-4rm8pb-MuiDivider-root"
    />
  </div>
</div>
`;

exports[`Cart - Item.mobile Should match snapshot with product 1`] = `
<div>
  <div
    class="MuiBox-root css-1m9t0zg"
  >
    <div
      class="MuiBox-root css-nqdwa1"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-x3cpnl"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fysqgs-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                <img
                  alt="1/2\\" Wrot Copper Cap"
                  class="MuiBox-root css-idsjlm"
                  src="http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-kfrp3-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1unv2py-MuiButtonBase-root-MuiIconButton-root"
              data-testid="remove-item-button-0"
              tabindex="0"
              type="button"
            >
              <svg>
                delete-bin.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-pad2pl-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1z0dvx8-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-rk7jif-MuiTypography-root"
            data-testid="cart-item-manufacturer-name-0"
          >
            Generic
          </span>
          <div>
            <p
              aria-label="1/2\\" Wrot Copper Cap"
              class="MuiTypography-root MuiTypography-body1 css-4mc1p7-MuiTypography-root"
              data-mui-internal-clone-element="true"
              data-testid="cart-item-product-name-0"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
                href="/product/generic-1-2-wrot-copper-cap-ccapd/MSC-33860"
              >
                1/2" Wrot Copper Cap
              </a>
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-zex3r6-MuiTypography-root"
              data-testid="cart-item-part-number-0"
            >
              Part#
               
              33860
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-qagaf3-MuiTypography-root"
              data-testid="cart-item-manufacturer-number-0"
            >
              MFR#
               
              CCAPD
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-18wiyal-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1j3agu2"
          >
            <div
              class="MuiBox-root css-yvkhq8"
            >
              <div
                class="number-input__container full-width"
              >
                <div
                  class="number-input__button-container small"
                >
                  <button
                    class="number-input__button"
                    data-testid="quantity-input-down-0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="RemoveIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13H5v-2h14v2z"
                      />
                    </svg>
                  </button>
                </div>
                <input
                  class="number-input__input small"
                  data-testid="quantity-input-0"
                  index="0"
                  min="1"
                  value="1"
                />
                <div
                  class="number-input__button-container small"
                >
                  <button
                    class="number-input__button"
                    data-testid="quantity-input-up-0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-2a4e9h-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-16lfl4z-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
            >
              Price
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
            data-testid="cart-item-unit-price-0"
          >
            $0.43 ea
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-4 css-11ojwqs-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1xrvs54-MuiTypography-root"
            >
              Available Qty
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1efrlbf-MuiTypography-root"
            data-testid="cart-item-qty-available-0"
          >
            215
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ecxygc-MuiTypography-root"
          >
            in stock
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-4 css-1bay0si-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1ebnygn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-6b5z9k-MuiTypography-root"
            >
              Order Total
            </span>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1fmbqzj-MuiTypography-root"
            data-testid="cart-item-total-price-0"
          >
            $0.43
          </h5>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiBox-root css-4rm8pb-MuiDivider-root"
    />
  </div>
</div>
`;
